src/api/v4/layer/metadata/base.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Base metadata object
+ *
+ * @constructor
+ * @abstract
+ * @memberof carto.layer.metadata
+ * @api
+ */
+function Base (type, rule) {
+  this._type = type || '';
+  this._column = rule.getColumn();
+  this._mapping = rule.getMapping();
+  this._property = rule.getProperty();
+}
+
+/**
+ * Return true is the metadata type is categories
+ *
+ * @return {boolean}
+ * @api
+ */
+Base.prototype.isCategories = function () {
```

Comment:
With this we are coupling the base class to its children.

IMO we should add an empty `.getType()` method in the base class and make each child override it returning an string.

`buckets.getType(); // 'buckets`
`categories.getType(); // 'categories`

This way we preserve the original functionality without forcing the base class to know its children, plus we get the ability to do switch sentences:

```javascript
switch(metadata.getType()) {
  case 'buckets':
    return handlebuckets(metadata);
  case 'categories':
    return handleCategories(metadata):
...
}
```